internal/services/cognitive/cognitive_account_rai_policy_resource.go

```diff
@@ -0,0 +1,396 @@
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+
+			id, err := raipolicies.ParseRaiPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			cognitiveAccountId := raipolicies.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
+
+			locks.ByID(cognitiveAccountId.ID())
+			defer locks.UnlockByID(cognitiveAccountId.ID())
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return raipolicies.ValidateRaiPolicyID
+}
+
+func expandRaiPolicyContentFilters(filters []AccountRaiPolicyContentFilter) *[]raipolicies.RaiPolicyContentFilter {
+	if filters == nil {
+		return nil
+	}
+
+	contentFilters := make([]raipolicies.RaiPolicyContentFilter, 0, len(filters))
+	for _, filter := range filters {
+		contentFilters = append(contentFilters, raipolicies.RaiPolicyContentFilter{
+			Name:              pointer.To(filter.Name),
+			Enabled:           pointer.To(filter.FilterEnabled),
+			Blocking:          pointer.To(filter.BlockEnabled),
+			SeverityThreshold: pointer.To(raipolicies.ContentLevel(filter.SeverityThreshold)),
+			Source:            pointer.To(raipolicies.RaiPolicyContentSource(filter.Source)),
+		})
+	}
+	return &contentFilters
+}
+
+func expandCustomBlockLists(list []AccountRaiPolicyCustomBlock) *[]raipolicies.CustomBlocklistConfig {
+	if list == nil {
+		return nil
+	}
+
+	customBlockLists := make([]raipolicies.CustomBlocklistConfig, 0, len(list))
+	for _, block := range list {
+		customBlockLists = append(customBlockLists, raipolicies.CustomBlocklistConfig{
+			BlocklistName: pointer.To(block.Name),
+			Blocking:      pointer.To(block.BlockEnabled),
+			Source:        pointer.To(raipolicies.RaiPolicyContentSource(block.Source)),
+		})
+	}
+	return &customBlockLists
+}
+
+func flattenRaiCustomBlockLists(blocklists *[]raipolicies.CustomBlocklistConfig) []AccountRaiPolicyCustomBlock {
+	if blocklists == nil {
+		return nil
+	}
+
+	customBlockLists := make([]AccountRaiPolicyCustomBlock, 0, len(*blocklists))
+	for _, block := range *blocklists {
+		customBlockLists = append(customBlockLists, AccountRaiPolicyCustomBlock{
+			Name:         pointer.From(block.BlocklistName),
+			BlockEnabled: pointer.From(block.Blocking),
+			Source:       string(pointer.From(block.Source)),
+		})
+	}
+	return customBlockLists
+}
+
+func flattenRaiPolicyContentFilters(filters *[]raipolicies.RaiPolicyContentFilter) []AccountRaiPolicyContentFilter {
+	if filters == nil {
+		return nil
+	}
+
+	contentFilters := make([]AccountRaiPolicyContentFilter, 0, len(*filters))
```

Comment:
We always need to set a value into state so this should be 
```suggestion
	contentFilters := make([]AccountRaiPolicyContentFilter, 0)
	if filters == nil {
		return contentFilters
	}

	
```